src/components/Tool/Appeal/Modals/EditAppealHeaderInfoModal/EditAppealHeaderInfoModal.tsx

```diff
@@ -0,0 +1,173 @@
+import React, { ReactElement } from 'react';
+import {
+  DialogActions,
+  DialogContent,
+  FormHelperText,
+  TextField,
+} from '@mui/material';
+import { Box } from '@mui/system';
+import { Formik } from 'formik';
+import { useSnackbar } from 'notistack';
+import { useTranslation } from 'react-i18next';
+import * as yup from 'yup';
+import { AppealFieldsFragment } from 'pages/accountLists/[accountListId]/tools/GetAppeals.generated';
+import { FieldWrapper } from 'src/components/Shared/Forms/FieldWrapper';
+import {
+  CancelButton,
+  SubmitButton,
+} from 'src/components/common/Modal/ActionButtons/ActionButtons';
+import Modal from 'src/components/common/Modal/Modal';
+import { useAccountListId } from 'src/hooks/useAccountListId';
+import {
+  AppealDocument,
+  AppealQuery,
+} from '../../AppealDetails/AppealsMainPanel/appealInfo.generated';
+import { useUpdateAppealMutation } from './EditAppeal.generated';
+
+interface EditAppealHeaderInfoModalProps {
+  handleClose: () => void;
+  appealInfo: AppealFieldsFragment;
+}
+
+export type EditAppealFormikSchema = {
+  name: string;
+  amount: number;
+};
+
+const EditAppealSchema: yup.SchemaOf<EditAppealFormikSchema> = yup.object({
+  name: yup.string().required('Please enter a name'),
+  amount: yup
+    .number()
+    .required('Please enter a goal')
+    .typeError('Appeal amount must be a valid number')
+    .test(
+      'Is positive?',
+      'Must use a positive number for appeal amount',
+      (value) => !value || parseFloat(value as unknown as string) > 0,
+    ),
+});
+
+export const EditAppealHeaderInfoModal: React.FC<
+  EditAppealHeaderInfoModalProps
+> = ({ appealInfo, handleClose }) => {
+  const { t } = useTranslation();
+  const accountListId = useAccountListId();
+  const { enqueueSnackbar } = useSnackbar();
+  const [UpdateAppeal] = useUpdateAppealMutation();
+
+  const onSubmit = async (attributes: EditAppealFormikSchema) => {
+    await UpdateAppeal({
+      variables: {
+        input: {
+          accountListId: accountListId ?? '',
+          attributes: {
+            id: appealInfo.id,
+            name: attributes.name,
+            amount: attributes.amount,
+          },
+        },
+      },
+      update: (cache) => {
+        const query = {
+          query: AppealDocument,
+          variables: {
+            accountListId: accountListId ?? '',
+            appealId: appealInfo.id,
+          },
+        };
+        const dataFromCache = cache.readQuery<AppealQuery>(query);
+
+        if (dataFromCache) {
+          const data = {
+            ...dataFromCache,
+            appeal: {
+              ...dataFromCache.appeal,
+              name: attributes.name,
+              amount: attributes.amount,
+            },
+          };
+          cache.writeQuery({ ...query, data });
```

Comment:
Did you know that when you have a pair of `readQuery`/`writeQuery` calls, you can use `updateQuery`? https://www.apollographql.com/docs/react/caching/cache-interaction/#using-updatequery-and-updatefragment

Reply:
Something I found I like as well is:

```
cache.modify({
          id: cache.identify({ __typename: 'Appeal', id: appealInfo.id }), // Identify the item in the cache
          fields: {
            name() {
              return attributes.name;
            },
            amount() {
              return attributes.amount;
            },
          },
        });
        ```